.github/workflows/image.yml

```diff
@@ -18,6 +18,10 @@ on: # yamllint disable-line rule:truthy
 jobs:
   image_build:
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+      id-token: write
```

Comment:
mind adding a note why these are needed?

Reply:
Yes, the id-token must be set to write in order for the OIDC to be used during the signing.  And, in GitHub Actions, when any permissions are specified, all others are set to none.  So, the other two permissions allow this action to read/write from registry.